go/oasis-test-runner/oasis/validator.go

```diff
@@ -101,19 +116,36 @@ func (net *Network) NewValidator(cfg *ValidatorCfg) (*Validator, error) {
 		net:           net,
 		dir:           valDir,
 		entity:        cfg.Entity,
+		sentries:      cfg.Sentries,
 		consensusPort: net.nextNodePort,
 		grpcDebugPort: net.nextNodePort + 1,
 	}
 
-	var valConsensusAddr node.Address
-	if err = valConsensusAddr.FromIP(netPkg.ParseIP("127.0.0.1"), val.consensusPort); err != nil {
-		return nil, fmt.Errorf("oasis/validator: failed to parse IP: %w", err)
+	var consensusAddrStr string
+	localhost := netPkg.ParseIP("127.0.0.1")
+	if len(val.sentries) > 0 {
+		var consensusAddrs []string
+		for _, sentry := range val.sentries {
+			var consensusAddr node.ConsensusAddress
+			consensusAddr.ID = sentry.publicKey
+			if err = consensusAddr.Address.FromIP(localhost, sentry.consensusPort); err != nil {
+				return nil, fmt.Errorf("oasis/validator: failed to parse IP address: %w", err)
+			}
+			consensusAddrs = append(consensusAddrs, consensusAddr.String())
+		}
+		consensusAddrStr = strings.Join(consensusAddrs, ",")
```

Comment:
might be nicer to pass multiple addresses be putting the `--node.consensus_address` flag multiple times? (and then using `viper.GetStringSlice` to get the slice of addresses)

Reply:
Indeed, the code already uses `viper.GetStringSlice` and I've made validator initialization pass the `--node.consensus_address` flag multiple times (if needed).